integration_test/gce/gce_testing.go

```diff
@@ -406,15 +407,25 @@ func hasNonEmptySeries(logger *log.Logger, it *monitoring.TimeSeriesIterator) (b
 // number of times. This is useful because it takes time for monitoring data to
 // become visible after it has been uploaded.
 func WaitForMetric(ctx context.Context, logger *log.Logger, vm *VM, metric string, window time.Duration, extraFilters []string) error {
```

Comment:
It seems like a waste to add a whole function and repeat the full description just to drop a return value. You could have just as easily made `WaitForMetric` return the `TimeSeries` and an error, and changed all the original call sites to drop the first value… The name would still fit…

Reply:
This was the result of some feedback on the design doc -- @martijnvans can weigh in, but the TL;DR was to achieve parity with `WaitForLog`/`QueryLog`. I'm also open to going the other way and achieving parity by consolidating both of them to just `WaitForLog` and `WaitForMetric`.

Reply:
Yeah, this comment could be applied to `QueryLog` as well. I'd vote for consolidating it in `WaitForLog` in the same way.

Reply:
I consolidated `WaitForMetric` for now. I need to think a bit more about `WaitForLog` because I just remembered that another reason we kept those separate is to add the `maxAttempts` parameter which is unused at most call sites.

Reply:
changing the call sites includes changing some files that live in google3. It's tricky to do that, so I suggested a new function. if we want, we can migrate the callsites in google3 to the new function and then delete the old function.

Reply:
simply giving waitforlog a new return value will break some tests in google3. That might be all right but I'd prefer to avoid it.

Reply:
Discussed offline; my preference is to add the return value to `WaitForMetric` and leave `WaitForLog` for a future refactor. There's only one callsite in google3 to change for `WaitForMetric`, which I can do after this is merged.